src/common.jl

```diff
@@ -1,194 +1,267 @@
-using DiffEqBase, DataStructures
+using DiffEqBase, OrdinaryDiffEq
+using StaticArrays: SVector, SizedArray
+using RecursiveArrayTools: ArrayPartition
 
-import DiffEqBase: ODEProblem, solve
+import DiffEqBase: ODEProblem, solve, ODE_DEFAULT_NORM, @.., addsteps!
 
-const warnkeywords =
-    (:save_idxs, :d_discontinuities, :unstable_check, :save_everystep,
-     :save_end, :initialize_save, :adaptive, :dt, :reltol, :dtmax,
-     :dtmin, :force_dtmin, :internalnorm, :gamma, :beta1, :beta2,
-     :qmax, :qmin, :qsteady_min, :qsteady_max, :qoldinit, :failfactor,
-     :maxiters, :isoutofdomain, :unstable_check,
-     :calck, :progress, :timeseries_steps, :tstops, :dense)
+import OrdinaryDiffEq: OrdinaryDiffEqAdaptiveAlgorithm,
+OrdinaryDiffEqConstantCache, OrdinaryDiffEqMutableCache,
+alg_order, alg_cache, initialize!, perform_step!, @unpack,
+@cache, stepsize_controller!, step_accept_controller!
+
+# TODO: check which keywords work fine
+const warnkeywords = (:save_idxs, :d_discontinuities, :unstable_check, :save_everystep,
+:save_end, :initialize_save, :adaptive, :dt, :reltol, :dtmax,
+:dtmin, :force_dtmin, :internalnorm, :gamma, :beta1, :beta2,
+:qmax, :qmin, :qsteady_min, :qsteady_max, :qoldinit, :failfactor,
+:isoutofdomain, :unstable_check,
+:calck, :progress, :timeseries_steps, :dense)
 
 global warnlist = Set(warnkeywords)
 
 
 
-abstract type TaylorAlgorithm <: DiffEqBase.DEAlgorithm end
+abstract type TaylorAlgorithm <: OrdinaryDiffEqAdaptiveAlgorithm end
 struct TaylorMethod <: TaylorAlgorithm
     order::Int
 end
+struct _TaylorMethod <: TaylorAlgorithm
+    order::Int
+    parse_eqs::Bool
+end
+
+_TaylorMethod(order; parse_eqs=true) = _TaylorMethod(order, parse_eqs) # set `parse_eqs` to `true` by default
+
+alg_order(alg::TaylorMethod) = alg.order
+alg_order(alg::_TaylorMethod) = alg.order
+
+# Regarding `isfsal`, thought about setting isfsal to false, but low order
+# interpolation may be helpful (since Taylor interpolation is memory-intensive),
+# so let's set isfsal to true for now, which is the default, so no method
+# definition needed here, so we in principle should support FSAL
+# isfsal(::TaylorMethod) = true # see discussion above
```

Comment:
Yet, this line is commented... Is it needed? We can certainly have it there commented, but then clarifying a bit would be very enlightening.

Reply:
It was more of a personal reminder when trying to figure the internals of DiffEqBase and OrdinaryDiffEq; I've now removed these commented lines